### DISCUSSION TITLE
How to pass data from a Client Component to Server Component

### DISCUSSION THREAD

**Discussion by pritsec** [vercel/next.js#58432] [App Router]
How to pass data from a Client Component to Server Component when we do router.push() in client component and we want to pass some sensitive data using some state management not by using router query parameter. So this is one of the very basis use case that we need to solve.

**Comment by prkagrawal** [comment]
Before changing routes, you can use [server actions](https://nextjs.org/docs/app/building-your-application/data-fetching/forms-and-mutations#how-server-actions-work).

Alternative [examples or proper ways to convert useState + useEffect to RSC + Next.js 13](https://github.com/vercel/next.js/discussions/47227#discussioncomment-5342500)

**Reply by pritsec** [reply]
This is not helpful, I need to send some data from client component to server component and based on the server component data I will do an api call. So this resource is not helpful for this scenario and cannot solve the problem
